### DISCUSSION TITLE
Is there a way to know how many events WebView_WebMessageReceived were attached to the webview2

### DISCUSSION THREAD

**Discussion by themhz** [MicrosoftEdge/WebView2Feedback#4254] [Q&A]
I have created a small app to test some javascript code directly on the webview2 https://github.com/themhz/WebView2JsDebugger .
anyway when I write some code say for example console.log("test"); and add the script to the webview2 and attach the event 
webView21.WebMessageReceived += WebView_WebMessageReceived;
nothing happens. 
if then I run something like that:
```
var data = { 'taskStatus': 'success', 'message': 'some data' }; 
window.chrome.webview.postMessage(JSON.stringify(data));
```
the event WebView_WebMessageReceived is added second time and then its fired twice. One for the previous execution event that was attached and one for the current. 
In general If I run some random js code say console.log("test"); and attach the event 3 times then if I run on time 
var data = { 'taskStatus': 'success', 'message': 'some data' }; 
window.chrome.webview.postMessage(JSON.stringify(data));  and attach the WebView_WebMessageReceived the fourth time then I get the message back 4 times. So it seems that this is being added to a que in the webview. I do remove the event by the following code when I receive the message:

```
private void WebView_WebMessageReceived(object sender, CoreWebView2WebMessageReceivedEventArgs args)
        {
            notify("WebMessageReceived event fired");
            string message = args.TryGetWebMessageAsString();
            webView21.WebMessageReceived -= WebView_WebMessageReceived;
            if (lstlog.Columns.Count > 0)
            {
                lstlog.Columns[0].Width = -2;
            }            
            lstlog.Items.Insert(0, message);           
        }
```

So the question: is there a way to know how many events were attached to the webview2 from with
webView21.WebMessageReceived += WebView_WebMessageReceived; ??

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/2944244/c6624cc4-5697-4ac9-8b13-42427c084894)
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/2944244/fb4adc31-b4b5-4397-b586-cf163f909a76)
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/2944244/9efd4003-1653-461e-a34b-171bb09ed33b)

This is the upload button code 

     ```
       private async void button1_Click(object sender, EventArgs e)
        {                        
            string script = scintilla1.Text;
            try
            {
                
                var parser = new JavaScriptParser();
                parser.ParseScript(script);
            }
            catch (ParserException scripterror)
            {
                MessageBox.Show($"Syntax error: {scripterror.Message}");
            }
                        
            
            if (webView21.CoreWebView2 != null)            
            {                
                if (ScriptChecker.checkIfPostMessageExists(script))
                {
                    // Handling web message received
                    webView21.WebMessageReceived += WebView_WebMessageReceived;
                }

                if (downloadStringEventAttacked == false)
                {
                    // Handling download event
                    webView21.CoreWebView2.DownloadStarting += CoreWebView2_DownloadStarting;
                    downloadStringEventAttacked = true;
                }
                
                // Execute the script
                var result = await webView21.ExecuteScriptAsync(script);
            }
            else
            {
                notify("You need to browse to a url to upload js code"); 
            }                                               
        }
```

**Comment by victorhuangwq** [comment]
@vbryh-msft is this something that is currently possible?

**Comment by vbryh-msft** [comment]
I'm not aware of any official ways of doing this and search engines are saying that it is expected for event handlers - can you accumulate number of calls to `webView21.WebMessageReceived += WebView_WebMessageReceived;` in some variable - kind of increment it before each call?
